src/script/lua_api/l_object.cpp

```diff
@@ -1885,6 +1885,61 @@ int ObjectRef::l_get_day_night_ratio(lua_State *L)
 	return 1;
 }
 
+// set_minimap_modes(self, modes, modeindex)
+int ObjectRef::l_set_minimap_modes(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	// Arg 1 should be a player
+	ObjectRef *ref = checkobject(L, 1);
+	RemotePlayer *player = getplayer(ref);
+	if (player == NULL)
+		return 0;
+
+	// Arg 2 should be a table (of tables)
+	if (!lua_istable(L, 2)) {
+		return 0;
+	}
+
+	// Arg 3 should be a number
+	s16 wantedmode = lua_tonumber(L, 3);
+
+	std::vector<MinimapMode> modes;
+
+	lua_pushnil(L);
+	while (lua_next(L, 2) != 0) {
+		/* key is at index -2, value is at index -1 */
+		if (lua_istable(L, -1)) {
+			MinimapMode mode;
+			std::string type = getstringfield_default(L, -1, "type", "");
+			if (type == "off")
+				mode.type = 0;
+			else if (type == "surface")
+				mode.type = 1;
+			else if (type == "radar")
+				mode.type = 2;
+			else if (type == "texture") {
+				mode.type = 3;
+				mode.extra = getstringfield_default(L, -1, "texture", "");
+			} else {
+				mode.type = -1;
+			}
+
+			if (mode.type >= 0) {
+				mode.label = getstringfield_default(L, -1, "label", "unknown");
+				mode.size = getintfield_default(L, -1, "size", 0);
```

Comment:
Size `513` and higher cause a segfault. Needs rangelim from 1 .. 512
```
(gdb) bt
#0  __memcmp_ssse3 () at ../sysdeps/x86_64/multiarch/memcmp-ssse3.S:1832
#1  0x0000555555bd0572 in Profiler::avg(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, float) ()
#2  0x00005555556b676d in ClientMap::renderMap(irr::video::IVideoDriver*, int) ()
#3  0x00007ffff76bf489 in irr::scene::CSceneManager::drawAll() ()
   from /usr/lib/x86_64-linux-gnu/libIrrlicht.so.1.8
#4  0x0000555555773dc1 in RenderingCore::draw3D() ()
#5  0x0000555555775a8d in RenderingCorePlain::drawAll() ()
#6  0x000055555570897c in Game::updateFrame(ProfilerGraph*, RunStats*, float, CameraOrientation const&) ()
```

Reply:
I tried with larger images (actually, just gave a try with a 2666x1971 jpeg map). Can you provide the image you used ? The problem could be due to something else than size.

Reply:
I was not able to get a segfault but anyway, with big size and big image, rendering performances gets very poor (it seems that the image->copyTo(map_image... is slow in that case). I'll add the rangelim.